src/xc_integrator/local_work_driver/device/cuda/kernels/uvvars.cu

```diff
@@ -209,4 +248,117 @@ void eval_uvvars_gga( size_t ntasks, size_t npts_total, int32_t nbf_max,
   );
 }
 
+
+
+
+
+
+
+
+
+
+template <int den_select>
+__global__ void eval_den_kern( size_t        ntasks,
+                                       XCDeviceTask* tasks_device ) {
+
+  const int batch_idx = blockIdx.z;
+  if( batch_idx >= ntasks ) return;
+
+  auto& task = tasks_device[ batch_idx ];
+
+  const auto npts            = task.npts;
+  const auto nbf             = task.bfn_screening.nbe;
+
+  double* den_eval_device   = nullptr;
+  // use the "U" variable (+/- for UKS) even though at this point the density (S/Z) is stored
+  switch( den_select ) {
```

Comment:
This is relying on the compiler to optimize around a switch statement. I'd imagine that if compiler optimization is off, this doesn't do what you want. I would work with `if constexpr` and potentially swapping `den_select` out to an `enum class` to avoid unintentional arguments (currently, if I pass `3` here, I'll get a segfault)